### PR TITLE
Replace speccy with redocly

### PR DESCRIPTION
**Pull request by nhsd-david-wass**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/components/stu3/schemas/responses/retrieveAdviceAndGuidanceWorklist/200Response.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ content:
     schema:
       $ref: '../../eRS-FetchWorklist-List.yaml'
     examples:
-      a&g-requests:
+      aAndg-requests:
         summary: Advice and guidance requests
```
Comment (nhsd-david-wass) [inline specification/components/stu3/schemas/responses/retrieveAdviceAndGuidanceWorklist/200Response.yaml:15]: '&' cause linting error - this field is not displayed - so does not affect output

File: redocly.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+extends:
+  - recommended
+rules:
+  no-invalid-media-type-examples:
+    severity: off
+  info-license-url: off
```
Comment (nhsd-david-wass) [inline redocly.yaml:6]: no-invalid-media-type-examples is a linting rule that should check all the examples against the  schema specification.
Where we have refs to example response files, the redocly library fails to expand the ref and fails the validation (known issue https://github.com/Redocly/redocly-cli/issues/919)

However, we have our own validation (validate_oas_examples.py) that validates all our examples against the schema

**Review by nhsd-jack-wainwright — COMMENTED** [review]
Looks good to me 🙂, just a couple very minor comments.

File: package.json [hunk 3]
```diff
@@ -3,16 +3,16 @@
...
+    "publish": "mkdir -p build && redocly bundle specification/e-referrals-service-api.yaml --dereferenced --remove-unused-components --ext json | poetry run python scripts/set_version.py | poetry run python scripts/populate_placeholders.py > build/e-referrals-service-api.json",
+    "serve": "redocly preview-docs -p 5000 build/e-referrals-service-api.json",
     "check-licenses": "node_modules/.bin/license-checker --failOn GPL --failOn LGPL"
   },
   "author": "NHS Digital",
   "license": "MIT",
   "homepage": "https://github.com/NHSDigital/e-referrals-service-api",
   "dependencies": {
-    "speccy": "^0.11.0"
+    "@redocly/cli": "^1.0.0-beta.124"
```
[the remark this answers is not in the document]
Comment (jeremyfiel) [reply package.json]: redocly now has a stable 1.0 release. https://github.com/Redocly/redocly-cli/releases/tag/%40redocly%2Fcli%401.2.0

**Pull request closed without merging**